addons/website_sale/controllers/main.py

```diff
@@ -257,7 +259,7 @@ def shop(self, page=0, category=None, search='', ppg=False, **post):
         products = search_product[offset: offset + ppg]
 
         ProductAttribute = request.env['product.attribute']
-        if products:
+        if products and ProductAttribute.search_count([]):
```

Comment:
I think you don't need count all the record of the table but know if exists a record

```suggestion
        if products and ProductAttribute.search([], limit=1, order='id'):
```

Reply:
changes applied, works fine.